File: web-sources/S8-fwks-palm/pages/simple/SimplePage.js

import { S8Object } from "/S8-api/S8Object.js";

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";


/**
 * 
 */
S8WebFront.CSS_import('/S8-fwks-palm/pages/simple/SimplePage.css');


/**
 * 
 */
export class SimplePage extends S8Object {


    /**
     * @type {S8Object[]}
     */
    elements;


    /**
     * 
     */
    selectionIndex = 1;

    
    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("simplepage");

        this.baseLayerNode = document.createElement("div");
        this.baseLayerNode.classList.add("simplepage-layer-base");
        this.wrapperNode.appendChild(this.baseLayerNode);
        
        this.modalLayerNode = document.createElement("div");
        this.modalLayerNode.classList.add("simplepage-layer-modal");
        this.modalLayerNode.style.display = "none";
        this.wrapperNode.appendChild(this.modalLayerNode);
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /**
     * 
     * @param {S8Object[]} elements 
     */
    S8_set_elements(elements){

        /* remove children */
        S8WebFront.removeChildren(this.baseLayerNode);

        let n = elements.length;
        for(let i=0; i<n; i++){
            let element = elements[i];
            this.baseLayerNode.appendChild(element.getEnvelope());
        }
        this.elements = elements;
    }


     /**
     * 
     * @param {ModalBox} elements 
     */
     S8_set_modalBox(box){
        if(box != null){
            this.modalLayerNode.appendChild(box.getEnvelope());
            this.modalLayerNode.style.removeProperty("display");
        }
        else{
            this.modalLayerNode.style.display = "none";
        }
    }


   /** continuous rendering approach */
    S8_render(){}

    /** continuous rendering approach */
    S8_dispose(){}
}

File: sources/com/s8/fwks/palm/components/pages/simple/SimplePage.java
package com.s8.fwks.palm.components.pages.simple;

import java.util.List;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.fwks.palm.components.PalmWebSources;
import com.s8.pkgs.ui.carbide.modal.box.ModalBox;


/**
 * 
 */
public class SimplePage extends S8WebObject {
	
	
	/**
	 * 
	 */
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/pages/simple/SimplePage";

	
	/**
	 * 
	 * @param front
	 */
	public SimplePage(S8WebFront front) {
		super(front, WEBPATH);
	}

	
	
	/**
	 * 
	 * @param menus
	 */
	public void setElements(List<S8WebObject> elements) {
		vertex.outbound().setObjectListField("elements", elements);
	}
	
	

	/**
	 * 
	 * @param menus
	 */
	public void setModalBox(ModalBox box) {
		vertex.outbound().setObjectField("modalBox", box);
	}
	

	public void publish() {
		vertex.expose(0);
	}
}
